src/lib.rs

```diff
@@ -71,16 +71,27 @@ pub async fn run<P: AsRef<Path> + TakeExt<PathBuf>>(
     println!("🤓  Intermediate checksums:");
     let mut prev_intermediate_checksums = String::new();
 
+    let checksums_intermediate_path = output_dir.join("checksums_intermediate.txt");
     File::options()
         .read(true)
         .write(true)
         .create(true)
-        .open(&output_dir.join("checksums_intermediate.txt"))?
-        .read_to_string(&mut prev_intermediate_checksums)?;
-    write_checksums(
-        &intermediate_wasm_paths,
-        &output_dir.join("checksums_intermediate.txt"),
-    )?;
+        .open(&checksums_intermediate_path)
+        .and_then(|mut file| file.read_to_string(&mut prev_intermediate_checksums))
+        .map_err(|err| {
+            anyhow::anyhow!(
+                "Failed read from \"{path}\":\n{err}",
+                path = checksums_intermediate_path.display(),
+                err = err,
```

Comment:
Until we have concrete Error types maybe we can just add:
```.context("Failed read from {checksums_intermediate_path}")``` instead?

Reply:
Thanks, updated it